Unable to compile qute codestart to native on windows

### Describe the bug

When I try compile resteasy-qute and rest-qute (resteasy-reactive-qute) codestart to native on windows the compilation fail with this error:
```
[1/8] Initializing...                                                                                   (11,5s @ 0,14GB)                                                                                                                                                        Java version: 21.0.2+13-LTS, vendor version: Oracle GraalVM 21.0.2+13.1                                                                                                                                                                                                        Graal compiler: optimization level: 2, target machine: x86-64-v3, PGO: ML-inferred                                                                                                                                                                                             C compiler: cl.exe (microsoft, x64, 19.38.33130)                                                                                                                                                                                                                               Garbage collector: Serial GC (max heap size: 80% of RAM)                                                                                                                                                                                                                       4 user-specific feature(s):                                                                                                                                                                                                                                                    - com.oracle.svm.thirdparty.gson.GsonFeature                                                                                                                                                                                                                                   - io.quarkus.runner.Feature: Auto-generated class by Quarkus from the existing extensions                                                                                                                                                                                      - io.quarkus.runtime.graal.DisableLoggingFeature: Disables INFO logging during the analysis phase                                                                                                                                                                              - org.eclipse.angus.activation.nativeimage.AngusActivationFeature                                                                                                                                                                                                             ------------------------------------------------------------------------------------------------------------------------                                                                                                                                                        3 experimental option(s) unlocked:                                                                                                                                                                                                                                             - '-H:+AllowFoldMethods' (origin(s): command line)                                                                                                                                                                                                                             - '-H:BuildOutputJSONFile' (origin(s): command line)                                                                                                                                                                                                                           - '-H:-UseServiceLoaderFeature' (origin(s): command line)                                                                                                                                                                                                                     ------------------------------------------------------------------------------------------------------------------------                                                                                                                                                       Build resources:                                                                                                                                                                                                                                                                - 12,09GB of memory (75,6% of 16,00GB system memory, determined at start)                                                                                                                                                                                                      - 2 thread(s) (100,0% of 2 available processor(s), determined at start)                                                                                                                                                                                                       [2/8] Performing analysis...  [*]                                                                       (76,8s @ 1,08GB)                                                                                                                                                          11 652 reachable types   (86,9% of   13 416 total)                                                                                                                                                                                                                             16 854 reachable fields  (58,6% of   28 750 total)                                                                                                                                                                                                                             63 411 reachable methods (58,6% of  108 179 total)                                                                                                                                                                                                                              3 693 types,   155 fields, and 3 238 methods registered for reflection                                                                                                                                                                                                             4 native libraries: crypt32, ncrypt, psapi, winhttp                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   ------------------------------------------------------------------------------------------------------------------------                                                                                                                                                                              11,3s (12,6% of total time) in 111 GCs | Peak RSS: 1,88GB | CPU load: 1,94                                                                                                                                                                              ------------------------------------------------------------------------------------------------------------------------                                                                                                                                                       Produced artifacts:                                                                                                                                                                                                                                                             C:\Users\jedla\Downloads\code-with-quarkus\target\code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar\svm_err_b_20240412T200838.238_pid2728.md (build_info)                                                                                                              ========================================================================================================================                                                                                                                                                       Failed generating 'code-with-quarkus-1.0.0-SNAPSHOT-runner' after 1m 28s.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     The build process encountered an unexpected error:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            com.oracle.svm.core.util.VMError$HostedError: InternalFeature defined by com.oracle.svm.hosted.ResourcesFeature unexpectedly failed with a(n) java.util.regex.PatternSyntaxException                                                                                                   at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:86)                                                                                                                                                                        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.handleFeatureError(FeatureHandler.java:287)                                                                                                                                                            at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:92)                                                                                                                                                                 at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$11(NativeImageGenerator.java:786)                                                                                                                                     at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.AbstractAnalysisEngine.runAnalysis(AbstractAnalysisEngine.java:181)                                                                                                                                              at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:783)                                                                                                                                               at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:592)                                                                                                                                                             at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:550)                                                                                                                                                               at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:539)                                                                                                                                            at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:721)                                                                                                                                                 at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:143)                                                                                                                                                 at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:98)                                                                                                                                           Caused by: java.util.regex.PatternSyntaxException: Unknown character property name {a} near index 11                                                                                                                                                                           templates\page.qute.html                                                                                                                                                                                                                                                                  ^                                                                                                                                                                                                                                                                           at java.base/java.util.regex.Pattern.error(Pattern.java:2204)                                                                                                                                                                                                                  at java.base/java.util.regex.Pattern.family(Pattern.java:3089)                                                                                                                                                                                                                 at java.base/java.util.regex.Pattern.sequence(Pattern.java:2332)                                                                                                                                                                                                               at java.base/java.util.regex.Pattern.expr(Pattern.java:2245)                                                                                                                                                                                                                   at java.base/java.util.regex.Pattern.compile(Pattern.java:1945)                                                                                                                                                                                                                at java.base/java.util.regex.Pattern.<init>(Pattern.java:1576)                                                                                                                                                                                                                 at java.base/java.util.regex.Pattern.compile(Pattern.java:1101)                                                                                                                                                                                                                at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ResourcesFeature.makeResourcePattern(ResourcesFeature.java:364)                                                                                                                                                       at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)                                                                                                                                                                                         at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)                                                                                                                                                                                         at java.base/java.util.concurrent.ConcurrentHashMap$KeySpliterator.forEachRemaining(ConcurrentHashMap.java:3573)                                                                                                                                                               at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)                                                                                                                                                                                             at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)                                                                                                                                                                                      at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)                                                                                                                                                                                             at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)                                                                                                                                                                                  at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)                                                                                                                                                                                            at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)                                                                                                                                                                                            at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)                                                                                                                                                                                             at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ResourcesFeature.compilePatterns(ResourcesFeature.java:357)                                                                                                                                                           at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ResourcesFeature.duringAnalysis(ResourcesFeature.java:335)                                                                                                                                                            at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$10(NativeImageGenerator.java:786)                                                                                                                                     at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:90)                                                                                                                                                                 ... 9 more
```

### Expected behavior

Finish build without any error.

### Actual behavior

```
Caused by: java.util.regex.PatternSyntaxException: Unknown character property name {a} near index 11
templates\page.qute.html
```

### How to Reproduce?

1. Download [https://code.quarkus.io/d?e=resteasy-qute&cn=code.quarkus.io](https://code.quarkus.io/d?e=resteasy-qute&cn=code.quarkus.io)
2. Compile to native by `mvn.cmd install -Dnative`
3. During the stage 2 error appear

### Output of `uname -a` or `ver`

Windows

### Output of `java -version`

_No response_

### Mandrel or GraalVM version (if different from Java)

GraalVM 23.1 JDK 21.0.2+13-LTS-jvmci-23.1-b30

### Quarkus version or git rev

3.8.3+, 3.9.0.CR2+

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache maven 3.8.8

### Additional information

Latest version where this is working are 3.8.2 and 3.9.0.CR1 (just tested if all 3.9 is affected)
Tried it on Windows 10 but this happening also on github runners. Tested it on Linux and build fine.

--- a/.github/native-tests.json
+++ b/.github/native-tests.json
@@ -133,9 +133,9 @@
             "os-name": "ubuntu-latest"
         },
         {
-            "category": "Windows - RESTEasy Jackson",
-            "timeout": 25,
-            "test-modules": "resteasy-jackson",
+            "category": "Windows support",
+            "timeout": 50,
+            "test-modules": "resteasy-jackson, qute",
             "os-name": "windows-latest"
         },
         {

--- a/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java
+++ b/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java
@@ -38,7 +38,6 @@ import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import jakarta.inject.Singleton;
 
@@ -93,12 +92,15 @@ import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.pkg.NativeConfig;
 import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
-import io.quarkus.fs.util.ZipUtils;
 import io.quarkus.gizmo.ClassOutput;
 import io.quarkus.gizmo.MethodDescriptor;
-import io.quarkus.maven.dependency.Dependency;
+import io.quarkus.maven.dependency.ArtifactKey;
+import io.quarkus.maven.dependency.DependencyFlags;
 import io.quarkus.maven.dependency.ResolvedDependency;
 import io.quarkus.panache.common.deployment.PanacheEntityClassesBuildItem;
+import io.quarkus.paths.FilteredPathTree;
+import io.quarkus.paths.PathFilter;
+import io.quarkus.paths.PathTree;
 import io.quarkus.qute.CheckedTemplate;
 import io.quarkus.qute.Engine;
 import io.quarkus.qute.EngineBuilder;
@@ -2124,9 +2126,6 @@ public class QuteProcessor {
             QuteConfig config,
             TemplateRootsBuildItem templateRoots)
             throws IOException {
-        Set<ApplicationArchive> allApplicationArchives = applicationArchives.getAllApplicationArchives();
-        List<ResolvedDependency> extensionArtifacts = curateOutcome.getApplicationModel().getDependencies().stream()
-                .filter(Dependency::isRuntimeExtensionArtifact).collect(Collectors.toList());
 
         // Make sure the new templates are watched as well
         watchedPaths.produce(HotDeploymentWatchedFileBuildItem.builder().setLocationPredicate(new Predicate<String>() {
@@ -2141,80 +2140,39 @@ public class QuteProcessor {
             }
         }).build());
 
-        for (ResolvedDependency artifact : extensionArtifacts) {
-            if (isApplicationArchive(artifact, allApplicationArchives)) {
-                // Skip extension archives that are also application archives
-                continue;
-            }
-            for (Path resolvedPath : artifact.getResolvedPaths()) {
-                if (Files.isDirectory(resolvedPath)) {
-                    scanRootPath(resolvedPath, config, templateRoots, watchedPaths, templatePaths,
-                            nativeImageResources);
-                } else {
-                    try (FileSystem artifactFs = ZipUtils.newFileSystem(resolvedPath)) {
-                        // Iterate over template roots, such as "templates", and collect the included templates
-                        for (String templateRoot : templateRoots) {
-                            Path artifactBasePath = artifactFs.getPath(templateRoot);
-                            if (Files.exists(artifactBasePath)) {
-                                LOGGER.debugf("Found template root in extension artifact: %s", resolvedPath);
-                                scanDirectory(artifactBasePath, artifactBasePath, templateRoot + "/", watchedPaths,
-                                        templatePaths,
-                                        nativeImageResources,
-                                        config);
-                            }
-                        }
-                    } catch (IOException e) {
-                        LOGGER.warnf(e, "Unable to create the file system from the path: %s", resolvedPath);
-                    }
-                }
+        final Set<ApplicationArchive> allApplicationArchives = applicationArchives.getAllApplicationArchives();
+        final Set<ArtifactKey> appArtifactKeys = new HashSet<>(allApplicationArchives.size());
+        for (var archive : allApplicationArchives) {
+            appArtifactKeys.add(archive.getKey());
+        }
+        for (ResolvedDependency artifact : curateOutcome.getApplicationModel()
+                .getDependencies(DependencyFlags.RUNTIME_EXTENSION_ARTIFACT)) {
+            // Skip extension archives that are also application archives
+            if (!appArtifactKeys.contains(artifact.getKey())) {
+                scanPathTree(artifact.getContentTree(), templateRoots, watchedPaths, templatePaths, nativeImageResources,
+                        config);
             }
         }
         for (ApplicationArchive archive : allApplicationArchives) {
-            archive.accept(tree -> {
-                for (Path root : tree.getRoots()) {
-                    // Note that we cannot use ApplicationArchive.getChildPath(String) here because we would not be able to detect
-                    // a wrong directory name on case-insensitive file systems
-                    scanRootPath(root, config, templateRoots, watchedPaths, templatePaths, nativeImageResources);
+            archive.accept(
+                    tree -> scanPathTree(tree, templateRoots, watchedPaths, templatePaths, nativeImageResources, config));
+        }
+    }
+
+    private void scanPathTree(PathTree pathTree, TemplateRootsBuildItem templateRoots,
+            BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
+            BuildProducer<TemplatePathBuildItem> templatePaths,
+            BuildProducer<NativeImageResourceBuildItem> nativeImageResources,
+            QuteConfig config) {
+        for (String templateRoot : templateRoots) {
+            pathTree.accept(templateRoot, visit -> {
+                if (visit != null) {
+                    // if template root is found in this tree then walk over its subtree
+                    scanTemplateRootSubtree(
+                            new FilteredPathTree(pathTree, PathFilter.forIncludes(List.of(templateRoot + "/**"))),
+                            visit.getPath(), watchedPaths, templatePaths, nativeImageResources, config);
                 }
             });
-        }
-    }
-
-    private void scanRootPath(Path rootPath, QuteConfig config, TemplateRootsBuildItem templateRoots,
-            BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
-            BuildProducer<TemplatePathBuildItem> templatePaths,
-            BuildProducer<NativeImageResourceBuildItem> nativeImageResources) {
-        scanRootPath(rootPath, rootPath, config, templateRoots, watchedPaths, templatePaths, nativeImageResources);
-    }
-
-    private void scanRootPath(Path rootPath, Path path, QuteConfig config, TemplateRootsBuildItem templateRoots,
-            BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
-            BuildProducer<TemplatePathBuildItem> templatePaths,
-            BuildProducer<NativeImageResourceBuildItem> nativeImageResources) {
-        if (!Files.isDirectory(path)) {
-            return;
-        }
-        try (Stream<Path> paths = Files.list(path)) {
-            for (Path file : paths.collect(Collectors.toList())) {
-                if (Files.isDirectory(file)) {
-                    // Iterate over the directories in the root
-                    // "/io", "/META-INF", "/templates", "/web", etc.
-                    Path relativePath = rootPath.relativize(file);
-                    if (templateRoots.isRoot(relativePath)) {
-                        LOGGER.debugf("Found templates root dir: %s", file);
-                        // The base path is an OS-specific template root path relative to the scanned root path
-                        String basePath = relativePath.toString() + relativePath.getFileSystem().getSeparator();
-                        scanDirectory(file, file, basePath, watchedPaths, templatePaths,
-                                nativeImageResources,
-                                config);
-                    } else if (templateRoots.maybeRoot(relativePath)) {
-                        // Scan the path recursively because the template root may be nested, for example "/web/public"
-                        scanRootPath(rootPath, file, config, templateRoots, watchedPaths, templatePaths, nativeImageResources);
-                    }
-                }
-            }
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
         }
     }
 
@@ -3427,55 +3385,24 @@ public class QuteProcessor {
                         readTemplateContent(originalPath, config.defaultCharset)));
     }
 
-    /**
-     *
-     * @param root
-     * @param directory
-     * @param basePath OS-specific template root path relative to the scanned root path, e.g. {@code templates/}
-     * @param watchedPaths
-     * @param templatePaths
-     * @param nativeImageResources
-     * @param config
-     * @throws IOException
-     */
-    private void scanDirectory(Path root, Path directory, String basePath,
+    private void scanTemplateRootSubtree(PathTree pathTree, Path templateRoot,
             BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
             BuildProducer<TemplatePathBuildItem> templatePaths,
             BuildProducer<NativeImageResourceBuildItem> nativeImageResources,
-            QuteConfig config)
-            throws IOException {
-        try (Stream<Path> files = Files.list(directory)) {
-            Iterator<Path> iter = files.iterator();
-            while (iter.hasNext()) {
-                Path filePath = iter.next();
-                /*
-                 * Fix for https://github.com/quarkusio/quarkus/issues/25751 where running tests in Eclipse
-                 * sometimes produces `/templates/tags` (absolute) files listed for `templates` (relative)
-                 * directories, so we work around this
-                 */
-                if (!directory.isAbsolute()
-                        && filePath.isAbsolute()
-                        && filePath.getRoot() != null) {
-                    filePath = filePath.getRoot().relativize(filePath);
+            QuteConfig config) {
+        pathTree.walk(visit -> {
+            if (Files.isRegularFile(visit.getPath())) {
+                LOGGER.debugf("Found template: %s", visit.getPath());
+                String templatePath = toOsAgnosticPath(templateRoot.relativize(visit.getPath()));
+                if (config.templatePathExclude.matcher(templatePath).matches()) {
+                    LOGGER.debugf("Template file excluded: %s", visit.getPath());
+                    return;
                 }
-                if (Files.isRegularFile(filePath)) {
-                    LOGGER.debugf("Found template: %s", filePath);
-                    Path relativePath = root.relativize(filePath);
-                    String templatePath = toOsAgnosticPath(relativePath);
-                    if (config.templatePathExclude.matcher(templatePath).matches()) {
-                        LOGGER.debugf("Template file excluded: %s", filePath);
-                        continue;
-                    }
-                    produceTemplateBuildItems(templatePaths, watchedPaths, nativeImageResources,
-                            basePath + relativePath.toString(),
-                            templatePath,
-                            filePath, config);
-                } else if (Files.isDirectory(filePath)) {
-                    LOGGER.debugf("Scan directory: %s", filePath);
-                    scanDirectory(root, filePath, basePath, watchedPaths, templatePaths, nativeImageResources, config);
-                }
+                produceTemplateBuildItems(templatePaths, watchedPaths, nativeImageResources,
+                        visit.getRelativePath(visit.getPath().getFileSystem().getSeparator()),
+                        templatePath, visit.getPath(), config);
             }
-        }
+        });
     }
 
     private static String toOsAgnosticPath(String path, FileSystem fs) {
@@ -3516,19 +3443,6 @@ public class QuteProcessor {
             }
             throw new IllegalStateException(builder.toString());
         }
-    }
-
-    private boolean isApplicationArchive(ResolvedDependency dependency, Set<ApplicationArchive> applicationArchives) {
-        for (ApplicationArchive archive : applicationArchives) {
-            if (archive.getKey() == null) {
-                continue;
-            }
-            if (dependency.getGroupId().equals(archive.getKey().getGroupId())
-                    && dependency.getArtifactId().equals(archive.getKey().getArtifactId())) {
-                return true;
-            }
-        }
-        return false;
     }
 
     static String readTemplateContent(Path path, Charset defaultCharset) {

--- a/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java
+++ b/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java
@@ -3355,31 +3355,29 @@ public class QuteProcessor {
      * @param templatePaths
      * @param watchedPaths
      * @param nativeImageResources
-     * @param osSpecificResourcePath The OS-specific resource path, i.e. templates\nested\foo.html
+     * @param resourcePath The relative resource path, including the template root
      * @param templatePath The path relative to the template root; using the {@code /} path separator
      * @param originalPath
      * @param config
      */
     private static void produceTemplateBuildItems(BuildProducer<TemplatePathBuildItem> templatePaths,
             BuildProducer<HotDeploymentWatchedFileBuildItem> watchedPaths,
-            BuildProducer<NativeImageResourceBuildItem> nativeImageResources, String osSpecificResourcePath,
+            BuildProducer<NativeImageResourceBuildItem> nativeImageResources, String resourcePath,
             String templatePath,
             Path originalPath, QuteConfig config) {
         if (templatePath.isEmpty()) {
             return;
         }
-        // OS-agnostic full path, i.e. templates/foo.html
-        String osAgnosticResourcePath = toOsAgnosticPath(osSpecificResourcePath, originalPath.getFileSystem());
         LOGGER.debugf("Produce template build items [templatePath: %s, osSpecificResourcePath: %s, originalPath: %s",
                 templatePath,
-                osSpecificResourcePath,
+                resourcePath,
                 originalPath);
         boolean restartNeeded = true;
         if (config.devMode.noRestartTemplates.isPresent()) {
-            restartNeeded = !config.devMode.noRestartTemplates.get().matcher(osAgnosticResourcePath).matches();
+            restartNeeded = !config.devMode.noRestartTemplates.get().matcher(resourcePath).matches();
         }
-        watchedPaths.produce(new HotDeploymentWatchedFileBuildItem(osAgnosticResourcePath, restartNeeded));
-        nativeImageResources.produce(new NativeImageResourceBuildItem(osSpecificResourcePath));
+        watchedPaths.produce(new HotDeploymentWatchedFileBuildItem(resourcePath, restartNeeded));
+        nativeImageResources.produce(new NativeImageResourceBuildItem(resourcePath));
         templatePaths.produce(
                 new TemplatePathBuildItem(templatePath, originalPath,
                         readTemplateContent(originalPath, config.defaultCharset)));
@@ -3399,7 +3397,7 @@ public class QuteProcessor {
                     return;
                 }
                 produceTemplateBuildItems(templatePaths, watchedPaths, nativeImageResources,
-                        visit.getRelativePath(visit.getPath().getFileSystem().getSeparator()),
+                        visit.getRelativePath("/"),
                         templatePath, visit.getPath(), config);
             }
         });

--- a/extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/templateroot/AdditionalTemplateRootTest.java
+++ b/extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/templateroot/AdditionalTemplateRootTest.java
@@ -51,11 +51,8 @@ public class AdditionalTemplateRootTest {
                         List<NativeImageResourceBuildItem> items = context.consumeMulti(NativeImageResourceBuildItem.class);
                         for (NativeImageResourceBuildItem item : items) {
                             if (item.getResources().contains("web/public/hello.txt")
-                                    || item.getResources().contains("web\\public\\hello.txt")
                                     || item.getResources().contains("templates/hi.txt")
-                                    || item.getResources().contains("templates\\hi.txt")
-                                    || item.getResources().contains("templates/nested/hoho.txt")
-                                    || item.getResources().contains("templates\\nested\\hoho.txt")) {
+                                    || item.getResources().contains("templates/nested/hoho.txt")) {
                                 found++;
                             }
                         }

